### PR TITLE
Sanitize user input

### PR DESCRIPTION
**Pull request by wez3**
Compiled and tested on raspberry pi. Please also whether all functionalities are still working :).

### CODE REVIEW

File: main/WebServer.cpp [hunk 1]
```diff
@@ -7893,11 +7893,21 @@ namespace http {
 			}
 
 			char szOrderBy[50];
-			if (order == "")
+			char szQuery[1000];
```
Comment (gizmocuz) [inline main/WebServer.cpp]: Could you make this a std::string ? we are not using sprintf

File: main/WebServer.cpp [hunk 2]
```diff
@@ -7893,11 +7893,21 @@ namespace http {
...
+			bool isAlpha = true;
+			const std::string orderBy = order.c_str();
+			for(int i = 0; i < orderBy.size(); i++) {
+				if( !isalpha(orderBy[i])) {
+					isAlpha = false;
+				}
+			}
+			if ((order == "") || (!isAlpha)) {
```
Comment (gizmocuz) [inline main/WebServer.cpp]: could you change this to 
`if ((order.empty()) || (!isAlpha)) {`

File: main/WebServer.cpp [hunk 3]
```diff
@@ -8129,15 +8140,16 @@ namespace http {
 						bAllowDeviceToBeHidden = true;
 					}
 
-					if (order == "")
+					if (order == "" || (!isAlpha))
```
Comment (gizmocuz) [inline main/WebServer.cpp]: could you change this to order.empty() instead of == ""

Comment (gizmocuz) [reply main/WebServer.cpp]: I know some other have been peeking/poking in this function as well, but i prefer checking against .empty()

File: main/WebServer.cpp [hunk 4]
```diff
@@ -8244,14 +8258,15 @@ namespace http {
...
```
Comment (gizmocuz) [inline main/WebServer.cpp]: could you change this to order.empty() instead of == ""

**Review by gizmocuz — COMMENTED** [review]
Thanks!! Just a few requests

File: main/WebServer.cpp (hunk 1 above)
Comment (wez3) [reply main/WebServer.cpp]: I am using strncpy later on to concat this variable. This function takes chars. Is there a equivalent function which takes std::string?

Comment (gizmocuz) [reply main/WebServer.cpp]: Yes, += you can just add strings.. Or I missed something here 🤔

**Comment by wez3** [comment]
Changed it to std::string.

**Comment by gizmocuz** [comment]
Thanks!!

**Pull request merged**